modules/mastodon/index.js

```diff
@@ -0,0 +1,320 @@
+      }
+    }
+  });
+}
+
+function htmlToText(s) {
+  return htmlToTextMod.fromString(s, {
+    wordwrap: null,
+    ignoreHref: true,
+  });
+}
+
+function formatRecord(record, fn) {
+  if (record.type == 'Note') {
+    fetchResource(record.attributedTo, (error, userRecord) => {
+      var str = '';
+      if (error) {
+        str += '(error getting user): ';
+      } else {
+        str += `${userRecord.name} (@${userRecord.preferredUsername}): `;
+      }
+      if (record.sensitive) {
+        str += sprintf('[content warning] %s', record.summary);
+      } else {
+        str += htmlToText(record.content);
+      }
+      fn(str);
+    });
+  } else if (record.type == 'Person') {
+    fn(`${record.name} (@${record.preferredUsername}): ` + htmlToText(record.summary));
+  }
+}
+
+module.exports.commands = ['masto'];
+
+module.exports.init = function(b) {
+  bot = b;
+  log = bot.log;
+
+  bot.readDataFile('config.json', function(err, data) {
+    if (err) {
+      log.info('initializing mastodon module data');
+      moduleConfig = defaultConfig();
+    } else {
+      try {
+        moduleConfig = JSON.parse(data);
+      } catch(ex) {
+        moduleConfig = defaultConfig();
+      }
+    }
+    writeConfig();
+  });
+
+  var userAgent = moduleConfig && moduleConfig.userAgent || 'EuIrcBot';
+  try {
+    var ver = require('../../package.json').version;
+    userAgent += '/' + ver;
+  } catch (ex) {
+    log.error('error getting package version: ' + ex);
+  }
+
+  var userAgentUrl = (moduleConfig && moduleConfig.userAgentUrl
+    || 'https://github.com/euank/EuIrcBot/blob/{commitish}/modules/mastodon/info.md');
+  if (userAgentUrl.includes('{commitish}')) {
+    var commitish = 'master';
+    try {
+      commitish = execSync('git rev-parse --short HEAD', {'encoding': 'utf8'}).trim();
+    } catch (ex) {
+      log.error('error determining deployed commit: ' + ex);
+    }
+    userAgentUrl = userAgentUrl.replace(/{commitish}/g, commitish);
+  }
+  userAgent += ' (+' + userAgentUrl + ')';
+
+  request = require('request').defaults({
+    headers: {'User-Agent': userAgent}
+  });
+};
+
+function handleUrl(url, reply) {
+  fetchResource(url, (error, record) => {
+    if (error) {
+      error.mastodonUrl = url;
+      log.info(error);
+      reply('error: ' + error.message);
+    } else {
+      formatRecord(record, (s) => reply.custom({ replaceNewlines: true }, s));
+    }
+  });
+
+  /*
+   * Instance autodiscovery: hit /api/v1/timelines/public and look for any new
+   * instance domains, then run those through forceCheckHostname to see if they
+   * ought to be added to our list.
+   *
+   * This will cause Pleroma instances to be added to the whitelist, even
+   * though they don't support ActivityPub objects, because they support the
+   * Mastodon client API.
+   */
+
+  robotRequest(
+    { url: urlMod.resolve(url, '/api/v1/timelines/public') },
+    (error, response, body) => {
+      if (error || response.statusCode != 200)
```

Comment:
nit, `statusCode >= 400` is more correct.

I could conceive of a mastodon server giving a `203 not authoritative` for content in some case eventually.

Reply:
Also nit, the style we've chosen and added an eslint config for since you started this requires a `{` for if statements, even one-liners.

This comment applies throughout. If you want to run `eslint` on your code with the eslintrc we've chosen, you can use `./node_modules/.bin/eslint <file>`.. but it won't be pretty. `--fix` can sometimes autofix some of it.